applications/TopologyOptimizationApplication/python_scripts/topology_optimizer_factory.py

```diff
@@ -0,0 +1,370 @@
+    def optimize(self):
+
+        print("\n> ==============================================================================================================")
+        print("> Starting optimization using the following algorithm: ",self.config.optimization_algorithm)
+        print("> ==============================================================================================================")
+
+        # Start timer and assign to object such that total time of opt may be measured at each step
+        self.opt_start_time = time.time()
+
+        # Initialize the design output in GiD format and print initial 0 state
+        self.gid_io.initialize_results(self.opt_model_part)
+        self.gid_io.write_results(0, self.opt_model_part, self.config.nodal_results, self.config.gauss_points_results)
+
+        # Call for the specified optimization algorithm
+        if(self.config.optimization_algorithm == "oc_algorithm"):
+           self.start_oc_algorithm()
+
+        elif(self.config.optimization_algorithm == "MMA_algorithm"):
+           self.start_mma_algorithm()
+
+        elif(self.config.optimization_algorithm == "GCMMA_algorithm"):
+           self.start_gcmma_algorithm()
+
+
+        else:
+            raise TypeError("Specified optimization_algorithm not implemented!")
+
+        # Finalize the design output in GiD format
+        self.gid_io.finalize_results()
+
+        # Stop timer
+        opt_end_time = time.time()
+
+        print("\n> ==============================================================================================================")
+        print("> Finished optimization in ",round(opt_end_time - self.opt_start_time,1)," s!")
+        print("> ==============================================================================================================")
+
+# ==============================================================================
+#           Topology Optimization with OC
+# ==============================================================================
+    # --------------------------------------------------------------------------
+    def start_oc_algorithm(self):
+
+        # Get Id of objective & constraint
+        only_F_id = None
+        only_C_id = None
+        for F_id in self.objectives:
+            only_F_id = F_id
+            break
+        for C_id in self.constraints:
+            only_C_id = C_id
+            break
+
+        # Initialize variables for comparison purposes in Topology Optimization Tool
+        pmax                          = self.config.penalty   # Maximum penalty value used for continuation strategy
+        Obj_Function                  = None
+        Obj_Function_old              = None
+        Obj_Function_initial          = None
+        Obj_Function_relative_change  = None
+        Obj_Function_absolute_change  = None
+
+        # Print the Topology Optimization Settings that will be used in the program
+        print("\n::[Topology Optimization Settings]::")
```

Comment:
Use logger

Reply:
@PhiHo-eng This is still open: please replace `print` in python code with the Kratos Logger.